processor/normalizesumsprocessor/processor.go

```diff
@@ -77,11 +74,38 @@ func (nsp *NormalizeSumsProcessor) transformMetrics(rms pdata.ResourceMetrics) {
 	}
 }
 
-// processMetric processes a Sum-type metric.
+func sliceContains(names []string, name string) bool {
+	for _, n := range names {
+		if name == n {
+			return true
+		}
+	}
+	return false
+}
+
+// processMetric processes a supported Sum-type or Gauge-type metric.
 // It returns a boolean that indicates if the metric should be kept.
 func (nsp *NormalizeSumsProcessor) processMetric(resource pdata.Resource, metric pdata.Metric) bool {
-	dps := metric.Sum().DataPoints()
+	if metric.DataType() == pdata.MetricDataTypeSum && metric.Sum().IsMonotonic() {
+		return nsp.processDataPoints(resource, metric, metric.Sum().DataPoints())
+	}
+	if metric.DataType() == pdata.MetricDataTypeGauge && sliceContains(nsp.includeGauges, metric.Name()) {
+		newMetric := pdata.NewMetric()
+		metric.CopyTo(newMetric)
+		newMetric.SetDataType(pdata.MetricDataTypeSum)
+		newMetric.Sum().SetIsMonotonic(true)
```

Comment:
Is there a processor we could just stick in front of this one to convert the targeted Gauges to Sums?

Reply:
+1, this doesn't seem like behavior that should be inside `normalizesumsprocessor`.

Reply:
Added a `casttosum` processor (though it could probably be called `casttocumulative`). PTAL.